### PR TITLE
0.9.0 errata, move to tmio for tuweni dep

### PR DESCRIPTION
**Pull request by garyschulte**
Remove tuweni as a runtime dependency for gnark-crypto, back to testImplementation dependency.  Also, bump tuweni dep to tmio.

In besu we have to specifically exclude this dependency from the gnark dep to prevent conflicts with besu's version of tuweni.

### CODE REVIEW

File: gnark/src/main/java/org/hyperledger/besu/nativelib/gnark/LibGnarkEIP196.java [hunk 1]
```diff
@@ -61,15 +60,23 @@ public static int eip196_perform_operation(
...
-      err_len.setValue(errBytes.size() - errBytes.numberOfTrailingZeroBytes());
+      err_len.setValue(LibGnarkUtils.findFirstTrailingZeroIndex(err));
       o_len.setValue(0);
     } else {
       err_len.setValue(0);
     }
     return ret;
   }
 
+  public int findFirstTrailingZeroIndex(byte[] array) {
```
Comment (pinges) [inline gnark/src/main/java/org/hyperledger/besu/nativelib/gnark/LibGnarkEIP196.java:71]: this is implemented in the Utils class. I think this is not needed ...
You are even using the Utils implementation further up :-)

**Review by pinges — CHANGES_REQUESTED** [review]
see comment

File: gnark/src/main/java/org/hyperledger/besu/nativelib/gnark/LibGnarkEIP196.java [hunk 2]
```diff
@@ -61,15 +60,23 @@ public static int eip196_perform_operation(
...
+    for (int i = array.length - 1; i >= 0; i--) {
+      if (array[i] != 0) {
+        return i + 1; // The first trailing zero is after this non-zero byte
+      }
+    }
+    // If all bytes are zero, return 0
+    return 0;
+  }
```
Comment (pinges) [inline gnark/src/main/java/org/hyperledger/besu/nativelib/gnark/LibGnarkEIP196.java:79]:
```suggestion
```

File: gnark/src/main/java/org/hyperledger/besu/nativelib/gnark/LibGnarkEIP196.java (hunk 1 above)
Comment (garyschulte) [reply gnark/src/main/java/org/hyperledger/besu/nativelib/gnark/LibGnarkEIP196.java]: where?  would love to DRY this out, but I don't see the usage you are talking about

Comment (garyschulte) [reply gnark/src/main/java/org/hyperledger/besu/nativelib/gnark/LibGnarkEIP196.java:71]: found and removed ;). thx

Comment (jflo) [reply gnark/src/main/java/org/hyperledger/besu/nativelib/gnark/LibGnarkEIP196.java:71]: This implementation has no usages, and can now be removed.

**Review by jflo — APPROVED** [review]
suggest removing removed method via removal

**Pull request merged**